app/models/guest_device.rb

```diff
@@ -10,4 +11,7 @@ class GuestDevice < ApplicationRecord
 
   has_one :network, :foreign_key => "device_id", :dependent => :destroy, :inverse_of => :guest_device
   has_many :miq_scsi_targets, :dependent => :destroy
+
+  has_many :firmwares, :dependent => :destroy
+  has_many :child_devices, -> { where(:parent_device_id => ids) }, :foreign_key => "parent_device_id", :class_name => "GuestDevice", :dependent => :destroy
```

Comment:
@blomquisg Could we just revert to my original changes (undo the change you recommended) and get this PR merged? Our provider changes for the network adapter support have already been merged but these changes have not. This partial implementation is causing the provider to not refresh properly. In addition, this PR is delaying the UI changes for network adapter support from being merged -- the PR is ready to go. We could always make another PR with your suggested changes and have it sit out there while someone investigates the database issues that occurring with the change. What do you think?